plugins/auto-sizes/tests/e2e/specs/improve-calculate-sizes.spec.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * External dependencies
+ */
+const path = require( 'path' );
+
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'check accurate sizes', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyfour' );
+		await requestUtils.deactivatePlugin( 'enhanced-responsive-images' );
+		await requestUtils.deactivatePlugin( 'modern-image-formats' );
+		await requestUtils.deleteAllMedia();
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllMedia();
+	} );
+
+	test( 'should get smaller version of image', async ( {
+		admin,
+		editor,
+		requestUtils,
+		page,
+	} ) => {
+		const filename = 'leaves.jpg';
+		const filepath = path.join(
+			__dirname + '/../../data/images/',
+			filename
+		);
+
+		await admin.createNewPost();
+		const media = await requestUtils.uploadMedia( filepath );
+
+		await editor.insertBlock( {
+			name: 'core/group',
+			attributes: {
+				align: 'wide',
+			},
+			innerBlocks: [
+				{
+					name: 'core/image',
+					attributes: {
+						alt: filename,
+						id: media.id,
+						url: media.source_url,
+					},
+				},
+			],
+		} );
+
+		const postId = await editor.publishPost();
+
+		// Navigate to the post and wait for the image to load.
+		await page.goto( `/?p=${ postId }` );
+		const imageElement = await page.getByAltText( filename );
+		const imageSizes = await imageElement.getAttribute( 'sizes' );
+
+		// Activate the plugin.
+		await requestUtils.activatePlugin( 'enhanced-responsive-images' );
+
+		// Reload the page and wait for the image to load.
+		await page.goto( `/?p=${ postId }` );
+		const updatedImageElement = await page.getByAltText( filename );
+
+		const updatedImageSizes =
+			await updatedImageElement.getAttribute( 'sizes' );
+
+		if ( imageSizes === updatedImageSizes ) {
+			throw new Error(
+				'Image sizes did not update after activating the plugin.'
+			);
+		}
```

Comment:
```suggestion
		expect( imageSizes ).toStrictEqual( updatedImageSizes );
```

Reply:
The `imageSizes` is not equal to `updatedImageSizes` that i update in https://github.com/WordPress/performance/pull/1988/commits/645ced5846208ce4fc96aa9ea73a5ae71b1cf7e1